### PR TITLE
Rdrp 856 filterout pnp fn

### PR DESCRIPTION
**Pull request by ewingr42**
## Pull Request submission

* changed filter_PNP_data helper function to reset index and drop the old one
* added test_filter_PNP_data unit test to test_staging_helpers    

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes # RDRP-856 

#### Code

- [ ] **Code runs** The code runs on my machine
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ y] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [y ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ y] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `requirements.txt`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/staging/staging_helpers.py [hunk 1]
```diff
@@ -445,10 +445,41 @@ def stage_validate_harmonise_postcodes(
...
     StagingHelperLogger.info("Finished PostCode Validation")
 
     return full_responses, postcode_mapper
+
+
+def filter_pnp_data(full_responses):
+    """
+    After the full_responses dataframe is created, filter out all PNP data
+    by simply excluding all records with legalstatus of 7.
+
```
Comment (AnneONS) [inline src/staging/staging_helpers.py]: The suggested style for docstrings is:
The first line should be in imperative form: Filter out all PNP data or equivalently records with legal status 7.

There should then be an empty line, and notes of explanation if needed. I don't think we need the extra info you've given here: the ACs for example could go in the pull request blurb.

File: src/staging/staging_helpers.py [hunk 2]
```diff
@@ -445,10 +445,41 @@ def stage_validate_harmonise_postcodes(
...
+    AC3 BERD separated data set is in a format that can be used during
+        the BERD pipeline run (a pandas dataframe and a CSV QA file)
+
+    Args:
+        full_responses (pandas.DataFrame):
+            The DataFrame containing the full resonses data.
+
+    Returns:
+        pandas.DataFrame: The filtered DataFrame
+        without rows where 'legalstatus' == '7'
```
Comment (AnneONS) [inline src/staging/staging_helpers.py]: can you either try and get this on a single line (88 characters max) or use the indent as you have above?

**Review by AnneONS — COMMENTED** [review]
Really great how you've got to grips with doing the unit test. 

Just a couple of minor comments here.

**Comment by ewingr42** [comment]
docstring has been tidied up...

tests rerun successfully

This change is part of the requirement to:
    AC1 BERD data is isolated from the full dataset
    AC3 BERD separated data set is in a format that can be used during
        the BERD pipeline run (a pandas dataframe and a CSV QA file)

**Pull request closed without merging**